libs/clients/signature-collection/src/lib/signature-collection.service.ts

```diff
@@ -0,0 +1,332 @@
+        min: user.svaedi?.fjoldi ?? 999,
+      },
+      signature:
+        user.medmaeli?.map((singature) => mapSignature(singature)) ?? [],
+      ownedLists: user.medmaelalistar
+        ? user.medmaelalistar?.map((list) => mapList(list))
+        : [],
+      isOwner: user.medmaelalistar ? user.medmaelalistar?.length > 0 : false,
+    }
+  }
+
+  async canCreate(nationalId: string): Promise<{ success: boolean }> {
+    // TODO: update when api returns correct data
+    const { canCreate } = await this.getSignee(nationalId)
+    return { success: canCreate }
+  }
+
+  async isOwner(nationalId: string): Promise<{ success: boolean }> {
+    const { isOwner } = await this.getSignee(nationalId)
+    return { success: isOwner }
+  }
+
+  async compareBulkSignaturesOnList(
+    listId: string,
+    nationalIds: string[],
+  ): Promise<Signature[]> {
+    // TODO: scope admin
+    // Takes a list of nationalIds listId and returns signatures found on list
+    const signaturesFound = await this.listsApi.medmaelalistarIDComparePost({
+      iD: parseInt(listId),
+      requestBody: nationalIds,
+    })
+    return signaturesFound.map(mapSignature)
+  }
+
+  async compareBulkSignaturesOnAllLists(
+    nationalIds: string[],
+  ): Promise<Signature[]> {
+    // TODO: scope admin
+    // Takes a list of nationalIds and returns signatures found on any list in current collection
+    const { id } = await this.currentCollectionInfo()
+    const signaturesFound =
+      await this.collectionsApi.medmaelasofnunIDComparePost({
+        iD: id,
+        requestBody: nationalIds,
+      })
+    return signaturesFound.map(mapSignature)
+  }
+
+  //   - DelegateList
+  // TODO: check if owner
+  //   - UndelegateList
+  // TODO: check if owner
+
+  async extendDeadline(listId: string, newEndDate: Date): Promise<List> {
+    // TODO: scope admin
+
+    const list = await this.listsApi.medmaelalistarIDExtendTimePatch({
+      iD: parseInt(listId),
+      newEndDate: newEndDate,
+    })
+    return mapList(list)
+  }
+
+  async bulkUploadSignatures(
+    listId: string,
+    nationalIds: string[],
+  ): Promise<BulkUpload> {
+    // TODO: scope admin
+
+    const signatures = await this.listsApi.medmaelalistarIDAddMedmaeliBulkPost({
+      iD: parseInt(listId),
+      requestBody: nationalIds,
+    })
+    return {
+      success:
+        signatures?.medmaeli?.map((signature) => mapSignature(signature)) ?? [],
+      failed: [
+        ...(signatures.notFound?.map((nationalId) => ({
+          nationalId,
+          reason: 'Kennitala fannst ekki',
+        })) ?? []),
+        ...(signatures.undirAldri?.map((nationalId) => ({
+          nationalId,
+          reason: 'Undir Aldri',
+        })) ?? []),
+        ...(signatures.ekkiASvaedi?.map((nationalId) => ({
+          nationalId,
+          reason: 'Ekki á svæði',
+        })) ?? []),
+        ...(signatures.ekkiIsRik?.map((nationalId) => ({
+          nationalId,
+          reason: 'ekkiIsRik',
```

Comment:
The other reasons have a clear icelandic text where as this one does not. Is this shown to a user?

Reply:
Has been changed to a better text